hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/client/HoodieFlinkWriteClient.java

```diff
@@ -152,6 +152,17 @@ public List<WriteStatus> upsert(List<HoodieRecord<T>> records, String instantTim
     return postWrite(result, instantTime, table);
   }
 
+  public List<WriteStatus> upsert(HoodieFlinkTable<T> table, List<HoodieRecord<T>> records, String instantTime) {
+    preWrite(instantTime, WriteOperationType.UPSERT, table.getMetaClient());
+    final HoodieWriteHandle<?, ?, ?, ?> writeHandle = getOrCreateWriteHandle(records.get(0), getConfig(),
+        instantTime, table, records.listIterator());
```

Comment:
There is no much sense to reuse the `table` object, we need a fresh new local timeline for the table each time we do the write action, and the timeline construction/refresh takes most of the time for table initialization.

So -1 on this change for correctness and robustness.

Reply:
> There is no much sense to reuse the `table` object, we need a fresh new local timeline for the table each time we do the write action, and the timeline construction/refresh takes most of the time for table initialization.
> 
> So -1 on this change for correctness and robustness.

From the flame graph, rebuilding the table initialization is time consuming, so I want to reuse this object as much as possible.